Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -821,6 +827,44 @@ const char * ObjectNode::GetObjExtension() const
 	return m_ObjExtensionOverride.Get();
 }
 
+// ContainsWarningsMSCL
+//------------------------------------------------------------------------------
+bool ObjectNode::ContainsWarningsMSCL(const char * data, uint32_t dataSize) const
+{
```

Comment:
If we extend DetermineFlags to check for /WX (warnings as errors) we could early out of this function:
```C++
// If /WX is enabled, compilation would have failed, so we can never have warnings
if ( GetFlag( FLAG_MSVC_WARNINGS_AS_ERRORS ) )
{
   return false; // cannot contain errors
}
```